Parser/Tree_Nodes/for_node.c

```diff
@@ -16,20 +16,72 @@
 #include "for_node.h"
 #include "b_stmt_list.h"
 
-for_node *create_for_node(asmt *initalizer, i_expr *conditional, r_asmt *incrementer, struct b_stmt_list *body) {
+
+for_node *create_for_node(GArray *token_stream, unsigned long index, unsigned long *next) {
     for_node *retVal = NULL;
+
+    // Make new for node
     retVal = calloc(1, sizeof(for_node));
-    retVal->initialize = initalizer;
-    retVal->conditional = conditional;
-    retVal->incrementer = incrementer;
-    retVal->body = body;
-    return retVal;
+    (*next)++;
+    Token *check = &g_array_index(token_stream, Token, *next);
```

Comment:
You should be using index, not next. Next is really only used after you create something with an unknown length. For example, when creating an assigmnment, we create the expression, then use next to make sure the following token is a semicolon